guides/source/wishlists.md

```diff
@@ -0,0 +1,1545 @@
+
+  has_many :subscribers, dependent: :destroy
+  has_many :wishlist_products, dependent: :destroy
+  has_many :wishlists, through: :wishlist_products
+  has_one_attached :featured_image
+  has_rich_text :description
+```
+
+We added two associations to `Product`. First, we associate the `Product` model
+with the `WishlistProduct` join table. Using this join table, our second
+association tells Rails that a `Product` is a part of many `Wishlists` through
+the `WishlistProduct` join table. From a `Product` record, we can directly
+access the `Wishlists` and Rails will know to automatically `JOIN` the tables in
+SQL queries.
+
+We also set `wishlist_products` as `dependent: :destroy`. When a `Product` is
+destroyed, it will be automatically removed from any Wishlists.
+
+In `app/models/wishlist.rb`, let's update both associations to enable counter
+caching:
+
+```ruby#2-5
+class WishlistProduct < ApplicationRecord
+  belongs_to :product, counter_cache: :wishlists_count
+  belongs_to :wishlist, counter_cache: :products_count
+
+  validates :product_id, uniqueness: { scope: :wishlist_id }
+end
+```
+
+We've specified a column name to update on the associated models. For the
+Product model, we want to use the `wishlists_count` column and for `Wishlist` we
+want to use `products_count`. These counter caches update anytime a
+`WishlistProduct` is created or destroyed.
+
+The uniqueness validation also tells Rails to check if a product is already on
+the wishlist. This is paired with the unique index on the wishlist_product table
+so that it's also validated at the database level.
+
+Finally, let's update `app/models/wishlist.rb` with it's associations:
+
+```ruby
+class Wishlist < ApplicationRecord
+  belongs_to :user
+  has_many :wishlist_products, dependent: :destroy
+  has_many :products, through: :wishlist_products
+end
+```
+
+Just like with `Product`, `wishlist_products` uses the `dependent: :destroy`
+option to automatically remove join table records when a Wishlist is deleted.
+
+### Friendly URLs
+
+Wishlists are often shared with friends and family. By default, the ID in the
+URL for a `Wishlist` is a simple Integer. This means we can't easily look at the
+URL to determine which `Wishlist` it's for.
+
+Active Record uses `to_param` method that can be used for generating pretty
+URLs. Let's try it out by creating the `to_param` method in our model:
+
+```ruby#6-8
+class Wishlist < ApplicationRecord
+  belongs_to :user
+  has_many :wishlist_products, dependent: :destroy
+  has_many :products, through: :wishlist_products
+
+  def to_param
+    "#{id}-#{name.squish.parameterize}"
+  end
+end
+```
+
+This returns a String for the URL param made up of the `id` and `name` joined by
+hyphens. `name` is made URL safe by using
+[`squish`](https://api.rubyonrails.org/classes/String.html#method-i-squish) to
+clean up whitespace and
+[`parameterize`](https://api.rubyonrails.org/classes/String.html#method-i-parameterize)
+to replace special characters.
```

Comment:
```suggestion
This will create a `to_param`  instance method that returns a String for the URL param made up of the `id` and `name` joined by
hyphens. `name` is made URL safe by using
[`squish`](https://api.rubyonrails.org/classes/String.html#method-i-squish) to
clean up whitespace and
[`parameterize`](https://api.rubyonrails.org/classes/String.html#method-i-parameterize)
to replace special characters.
```